pennylane/transforms/compile.py

```diff
@@ -182,9 +183,13 @@ def qfunc(x, y, z):
         basis_set = basis_set or all_ops
 
         def stop_at(obj):
+            if not isinstance(obj, qml.operation.Operator):
+                return True
             return obj.name in basis_set and (not getattr(obj, "only_visual", False))
 
-        expanded_tape = tape.expand(depth=expand_depth, stop_at=stop_at)
+        [expanded_tape], _ = qml.devices.preprocess.decompose(
+            tape, stopping_condition=stop_at, max_expansion=expand_depth, name="compile"
```

Comment:
Very minor thing, but if we are going to start using this in transforms and passing the transform name, maybe we could update the text in `_operator_decomposition_gen` to say `"Operator {op} not supported with {name}` rather than `on {name}`. I think that still makes sense if `name` is a device name, and makes more sense with the transform names.

Reply:
Switch `with` to `on`.

One of the main reasons I actually wanting it to move to the transforms folder is the error raised.  I want to raise a `DeviceError` when the transform is used in a device, but a `DecompositionUndefinedError` when used in other contexts 🤔

Reply:
And I want to use a `CompileError` when it's used with Catalyst 🙃 Maybe we should just make `error_type` a `kwarg` that defaults to `DecompositionUndefinedError`